packages/form/src/utils/__tests__/parseSelectOptions.test.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import SelectOption from 'src/SelectOption';
+import parseSelectOptions from '../parseSelectOptions';
+
+it('reads options from React children of <SelectOption>s', () => {
+    const singleOption = <SelectOption label="Foo" value="foo" />;
+
+    expect(parseSelectOptions(singleOption)).toMatchObject([
+        { label: 'Foo', value: 'foo' },
+    ]);
+
+    const multipleOptions = [
+        <SelectOption label="Foo" value="foo" checked />,
+        <SelectOption label="Bar" value="bar" readOnly />,
+        <SelectOption label="Meh" value="meh" />,
+    ];
+    expect(parseSelectOptions(multipleOptions)).toMatchObject([
+        { label: 'Foo', value: 'foo', checked: true },
+        { label: 'Bar', value: 'bar', readOnly: true },
+        { label: 'Meh', value: 'meh' },
+    ]);
+});
+
+it('ignores children that are not <SelectOption>', () => {
+    const children = [
+        <SelectOption label="Foo" value="foo" />,
+        'Hello World!',
```

Comment:
How about show a warning in console if containing non-react component in children?

Reply:
Yeah I'll add a todo there.

I was thinking that someone may want to put anything into `<SelectList>` through `children`, like the basic `<List>` takes anything from `children`.

But after a second thought, I don't think it makes any sense to allow anything except `<SelectOption>` to be passed into that.